Update docs-elastic-publish.yml

glitteringkatie opened the pull request:
Done as part of our serverless move (https://github.com/elastic/docs-projects/issues/178)

- Copies the public docs builder for a public repo (https://github.com/elastic/workflows/tree/main?tab=readme-ov-file#public-repos)
- Leaves all paths to consume from root

shainaraskas reviewed (approved):
workflow looks good - do you want me to update codeowners in a separate PR or do you want to do it as part of this one?

glitteringkatie commented on .github/workflows/docs-elastic-publish.yml:
```diff
@@ -1,25 +1,29 @@
-#       VERCEL_PROJECT_ID_DOCS_CO: ${{ secrets.VERCEL_PROJECT_ID_DOCS_CO }}
+jobs:
+  publish:
+    if: contains(github.event.pull_request.labels.*.name, 'ci:doc-build')
```
It's for security's sake! Because the repo is public and essentially you can write any javascript in mdx and make it work, it's safer to not auto-build on every PR. Plus, only maintainers (I believe) are able to put a label on a PR. So ideally, the label is only added once an elastician has confirmed the PR doesn't look fishy 🐟

glitteringkatie commented:
>do you want me to update codeowners in a separate PR or do you want to do it as part of this one?

Added! Also renamed the workflow so they are all consistent.

shainaraskas reviewed (dismissed):
perfect!

glitteringkatie commented on .github/workflows/co-docs-builder.yml:
```diff
@@ -0,0 +1,29 @@
+      VERCEL_GITHUB_TOKEN: ${{ secrets.VERCEL_GITHUB_TOKEN_PUBLIC }}
+      VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN_PUBLIC }}
+      VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID_PUBLIC }}
+      VERCEL_PROJECT_ID_DOCS_CO: ${{ secrets.VERCEL_PROJECT_ID_DOCS_CO }}
```
```suggestion
      VERCEL_PROJECT_ID_DOCS_CO: ${{ secrets.VERCEL_PROJECT_ID_DOCS_CO_PUBLIC }}
```

glitteringkatie commented on .github/CODEOWNERS:
```diff
@@ -1 +1,2 @@
 * @elastic/platform-docs
+/.github/workflows/ @elastic/docs-engineering
```
```suggestion
/.github/workflows/co-docs-builder.yml @elastic/docs-engineering
```

shainaraskas reviewed (approved):
good call making the ownership more specific!
